go-controller/pkg/cni/helper_linux.go

```diff
@@ -143,41 +143,59 @@ func setupSriovInterface(netns ns.NetNS, containerID, ifName string, ifInfo *Pod
 	}
 	vfNetdevice := vfNetdevices[0]
 
-	// 2. get Uplink netdevice
-	uplink, err := sriovnet.GetUplinkRepresentor(pciAddrs)
-	if err != nil {
-		return nil, nil, err
-	}
-
-	// 3. get VF index from PCI
+	// 2. get VF index from PCI
 	vfIndex, err := sriovnet.GetVfIndexByPciAddress(pciAddrs)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	// 4. lookup representor
-	rep, err := sriovnet.GetVfRepresentor(uplink, vfIndex)
-	if err != nil {
-		return nil, nil, err
-	}
-	oldHostRepName := rep
+	if !ifInfo.IsSmartNic {
 
-	// 5. rename the host VF representor
-	hostIface.Name = containerID[:15]
-	if err = renameLink(oldHostRepName, hostIface.Name); err != nil {
-		return nil, nil, fmt.Errorf("failed to rename %s to %s: %v", oldHostRepName, hostIface.Name, err)
-	}
-	link, err := netlink.LinkByName(hostIface.Name)
-	if err != nil {
-		return nil, nil, err
-	}
-	hostIface.Mac = link.Attrs().HardwareAddr.String()
+		// 3. get Uplink netdevice
+		uplink, err := sriovnet.GetUplinkRepresentor(pciAddrs)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		// 4. lookup representor
+		rep, err := sriovnet.GetVfRepresentor(uplink, vfIndex)
+		if err != nil {
+			return nil, nil, err
+		}
+		oldHostRepName := rep
+
+		// 5. rename the host VF representor
+		hostIface.Name = containerID[:15]
+		if err = renameLink(oldHostRepName, hostIface.Name); err != nil {
+			return nil, nil, fmt.Errorf("failed to rename %s to %s: %v", oldHostRepName, hostIface.Name, err)
+		}
+		rep_link, err := netlink.LinkByName(hostIface.Name)
+		if err != nil {
+			return nil, nil, err
+		}
+		hostIface.Mac = rep_link.Attrs().HardwareAddr.String()
 
-	// 6. set MTU on VF representor
-	if err = netlink.LinkSetMTU(link, ifInfo.MTU); err != nil {
-		return nil, nil, fmt.Errorf("failed to set MTU on %s: %v", hostIface.Name, err)
+		// 6. set MTU on VF representor
+		if err = netlink.LinkSetMTU(rep_link, ifInfo.MTU); err != nil {
+			return nil, nil, fmt.Errorf("failed to set MTU on %s: %v", hostIface.Name, err)
+		}
 	}
 
+	//@TODO need to check why it failed on BF (it probeley need to be with TC)
+	/*
```

Comment:
do you know why this is commented out ? can you check on the system what fails in these steps (just invoke ip link commands and see what happens first)

Reply:
It is failing on the BF host with operation not permitted: 
[root@bf-node-worker0 ~]# ip link set enp6s0f0 vf 3 max_tx_rate 0
RTNETLINK answers: Operation not permitted
[root@bf-node-worker0 ~]# ip link set enp6s0f0 vf 3 min_tx_rate 0
RTNETLINK answers: Operation not permitted

For Non BF devices it is working

I will keep it there for now as a note to fix it later

Reply:
dmesg says anything ?

Reply:
no